src/test/BeefyEpochStrategy.t.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+import {TestPlus} from "./TestPlus.sol";
+import {stdStorage, StdStorage} from "forge-std/Test.sol";
+import {Deploy} from "./Deploy.sol";
+
+import {StrategyVault} from "src/vaults/locked/StrategyVault.sol";
+import {BeefyEpochStrategy} from "src/strategies/BeefyEpochStrategy.sol";
+
+import {BaseStrategy} from "src/strategies/BaseStrategy.sol";
+import {ICurvePool, I3CrvMetaPoolZap} from "src/interfaces/curve.sol";
+
+import {IBeefyVault} from "src/interfaces/Beefy.sol";
+
+import {DeployLib} from "script/ConvexStrategy.s.sol";
+
+import {console} from "forge-std/console.sol";
+
+contract TestBeefyWithStrategyVault is TestPlus {
+    StrategyVault vault;
+    ERC20 usdc = ERC20(0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174);
+    ICurvePool pool = ICurvePool(0xa138341185a9D0429B0021A11FB717B225e13e1F);
+    I3CrvMetaPoolZap zapper = I3CrvMetaPoolZap(0x5ab5C56B9db92Ba45a0B46a207286cD83C15C939);
+    IBeefyVault beefy = IBeefyVault(0x2520D50bfD793D3C757900D81229422F70171969);
+    ERC20 curveLPToken = ERC20(0xa138341185a9D0429B0021A11FB717B225e13e1F);
+
+    BeefyEpochStrategy strategy;
+
+    uint256 initialAssets;
+
+    function setupBeefyStrategy() public {
+        address[] memory strategists = new address[](1);
+        strategists[0] = address(this);
+        strategy = new BeefyEpochStrategy(
+            vault, 
+            pool,
+            zapper,
+            2, // assetIndex
+            beefy,
+            strategists
+        );
+    }
+
+    function setUp() public {
+        vm.createSelectFork("polygon", 44_367_000);
+        StrategyVault sVault = new StrategyVault();
+
+        bytes memory initData =
+            abi.encodeCall(StrategyVault.initialize, (governance, address(usdc), "BeefyVault", "BeefyVault"));
+
+        vm.prank(governance);
+        ERC1967Proxy proxy = new ERC1967Proxy(address(sVault), initData);
```

Comment:
I see that your testing the upgrade below. Once that's done we can remove the use of the proxy contract.